packages/@lwc/ssr-compiler/src/compile-template/transformers/component/slotted-content.ts

```diff
@@ -75,6 +75,17 @@ const bAddLightContent = esTemplate`
     });
 `<EsCallExpression>;
 
+function getShadowSlottedContent(slottableChildren: IrChildNode[], cxt: TransformerContext) {
+    return optimizeAdjacentYieldStmts(
+        slottableChildren.flatMap((child) => {
+            if (child.type === 'ExternalComponent') {
+                cxt.isSlotted = false;
+            }
+            return irToEs(child, cxt);
+        })
+    );
+}
+
```

Comment:
I kept the solution here as minimal as possible on purpose. It's likely we're missing some tests.